openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -0,0 +1,98 @@
+"""
+Provides partition support to the user service.
+"""
+import logging
+from course_modes.models import CourseMode
+from student.models import CourseEnrollment
+from verify_student.models import SkippedReverification, VerificationStatus
+
+log = logging.getLogger(__name__)
+
+
+class VerificationPartitionScheme(object):
+    """
+    This scheme randomly assigns users into the partition's groups.
+    """
+    NON_VERIFIED_GROUP = 'non_verified_group'
+    VERIFIED_ALLOW_GROUP = 'verified_allow_group'
+    VERIFIED_DENY_GROUP = 'verified_deny_group'
+
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition):
+        checkpoint = user_partition.parameters["location"]
```

Comment:
Yes, we should be defensive.  A fairly safe default behavior would be to log an error with the course key and the partition ID (so we can find it), then put the user in a default group.  `NON_VERIFIED` probably makes sense as a default, so at least we won't accidentally prevent students from accessing content.